src/initilization-context.ts

```diff
@@ -48,6 +66,6 @@ export class InitializationContext {
 	}
 
 	get readyPromise() {
```

Comment:
The type in the name kind of bothered me, but its only used internally, so not worth worrying about.

Reply:
Yeah, it was originally `readyPromise` so as not to conflict with the `ready` method. I can change it now though.

Reply:
Cool. I think `context.ready.then(...)` reads pretty well.

Reply:
`whenReady` can potentially execute the callback immediately/synchronously, which is desirable in the event initialization has no async operations (calling an entity constructor). If everything used the promise, then it would default to async.

Reply:
Makes sense 👍